ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/LedgerWriter.scala

```diff
@@ -27,8 +27,18 @@ trait LedgerWriter extends ReportsHealth {
     *
     * @param correlationId correlation ID to be used for logging purposes
     * @param envelope      opaque submission; may be compressed
+    * @param metadata      metadata associated to this particular commit
     * @return future for sending the submission; for possible results see
     *         [[com.daml.ledger.participant.state.v1.SubmissionResult]]
     */
-  def commit(correlationId: String, envelope: Bytes): Future[SubmissionResult]
+  def commit(
+      correlationId: String,
+      envelope: Bytes,
+      metadata: CommitMetadata): Future[SubmissionResult] = commit(correlationId, envelope)
+
+  // @deprecated("Will be removed in 1.4.0", "1.3.0")
+  def commit(
+      correlationId: String,
+      envelope: Bytes
+  ): Future[SubmissionResult] = ???
```

Comment:
Shouldn't this be delegating to the above?

Reply:
Yes, changed -- just was checking if it is being called from anywhere.